### PR TITLE
Add support for all proxy schemes, not just https://

### PR DESCRIPTION
**Pull request by canth1**
This commit fixes https://github.com/puppetlabs/puppetlabs-apache/issues/2285 by supporting all mod_proxy schemes according to the documentation here:

https://httpd.apache.org/docs/2.4/mod/mod_proxy.html

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like the tests fail on this.

I also wonder if this should be a defined type.

**Comment by david22swan** [comment]
> Looks like the tests fail on this.
> 
> I also wonder if this should be a type alias.

I would agree on this

**Review by david22swan — CHANGES_REQUESTED** [review]
@canth1 
Hey, look's like you've got a few failing tests

Also, think that @ekohl suggestion is a good one and that this would be a good use of a type alias to keep the code neater

File: types/modproxy.pp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# @summary Supported protocols / schemes by mod_proxy
+# @see https://httpd.apache.org/docs/2.4/mod/mod_proxy.html
+type Apache::ModProxyProtocol = Pattern[/(\A(ajp|fcgi|ftp|h2|h2c|http|https|scgi|uwsgi|ws|wss):\/\/.*\z)/, /(\Aunix:\/([^\n\/\0]+\/*)*\z)/]
```
Comment (ekohl) [inline types/modproxy.pp:3]: Is it valid to not specify anything or should it be
```suggestion
type Apache::ModProxyProtocol = Pattern[/(\A(ajp|fcgi|ftp|h2|h2c|http|https|scgi|uwsgi|ws|wss):\/\/.+\z)/, /(\Aunix:\/([^\n\/\0]+\/*)*\z)/]
```

**Review by ekohl — APPROVED** [review]
Small detail, but this looks good.

**Review by ekohl — CHANGES_REQUESTED** [review]
Looks like the tests are failing.

Also, it's possible to add tests for a type alias: https://rspec-puppet.com/documentation/type_aliases/

**Comment by canth1** [comment]
> Small detail, but this looks good.

I don't think that's correct, because I took my regex directly from Stdlib:HTTPSUrl

https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/httpsurl.pp

Which is:

type Stdlib::HTTPSUrl = Pattern[/(?i:\Ahttps:\/\/.*\z)/]

**Comment by canth1** [comment]
> Looks like the tests are failing.
> 
> Also, it's possible to add tests for a type alias: https://rspec-puppet.com/documentation/type_aliases/

Ok, I'll add the tests for this new type alias.

**Comment by ekohl** [comment]
> > Small detail, but this looks good.
> 
> I don't think that's correct, because I took my regex directly from Stdlib:HTTPSUrl
> 
> https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/httpsurl.pp
> 
> Which is:
> 
> type Stdlib::HTTPSUrl = Pattern[/(?i:\Ahttps://.*\z)/]

Interesting. That implies it also accepts `https://` which doesn't make sense to me since I don't consider it an HTTP(S) URL.

File: types/modproxy.pp [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# @summary Supported protocols / schemes by mod_proxy
+# @see https://httpd.apache.org/docs/2.4/mod/mod_proxy.html
+type Apache::ModProxyProtocol = Pattern[/(\A(ajp|fcgi|ftp|h2c?|https?|scgi|uwsgi|wss?):\/\/.*\z)/, /(\Aunix:\/([^\n\/\0]+\/*)*\z)/]
```
Comment (ekohl) [inline types/modproxy.pp:3]: I don't think this is correct. The filename must match the type name.

**Comment by canth1** [comment]
> > > Small detail, but this looks good.
> > 
> > 
> > I don't think that's correct, because I took my regex directly from Stdlib:HTTPSUrl
> > https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/types/httpsurl.pp
> > Which is:
> > type Stdlib::HTTPSUrl = Pattern[/(?i:\Ahttps://.*\z)/]
> 
> Interesting. That implies it also accepts `https://` which doesn't make sense to me since I don't consider it an HTTP(S) URL.

Actually, you're correct. We shouldn't allow an empty value after ://

 I'll fix that with your suggestion.

It looks like the Stdlib:HTTPSUrl allows that for some reason?

**Comment by ekohl** [comment]
> It looks like the Stdlib:HTTPSUrl allows that for some reason?

Possibly an oversight and nobody ran into problems because of it.

**Comment by canth1** [comment]
The spec tests are failing, but I can't figure out why?

Resource type not found: Apache::ModProxyProtocol

Not sure why it's giving me that error?

**Comment by ekohl** [comment]
Because of https://github.com/puppetlabs/puppetlabs-apache/pull/2289#discussion_r948196017. You must name it according to the conventions so the loader finds it, so `types/mod_proxy_protocol.pp`.

**Review by ekohl — APPROVED** [review]
This looks good to me. @david22swan you requested changes, mind taking another look? Also, IMHO we should squash this on merge.

**Review by david22swan — APPROVED** [review]
LGTM

Everything look's good to me, only small comment is that the variables aren't aligned, but since there are only two of them it's not that big of a deal in my eyes.
Will go ahead with a squash merge.

**Pull request merged**